metron-analytics/metron-statistics/src/test/java/org/apache/metron/statistics/StellarStatisticsFunctionsTest.java

```diff
@@ -356,6 +357,44 @@ public void testSkewness() throws Exception {
     assertEquals(stats.getSkewness(), (Double) actual, 0.1);
   }
 
+  @Test
+  public void testStatsBin() throws Exception {
+    statsInit(windowSize);
+    statsBinRunner(StellarStatisticsFunctions.Bin.BinSplits.QUARTILE.split);
+    statsBinRunner(StellarStatisticsFunctions.Bin.BinSplits.QUARTILE.split, "'QUARTILE'");
+    statsBinRunner(StellarStatisticsFunctions.Bin.BinSplits.QUINTILE.split, "'QUINTILE'");
+    statsBinRunner(StellarStatisticsFunctions.Bin.BinSplits.DECILE.split, "'DECILE'");
+    statsBinRunner(ImmutableList.of(25.0, 50.0, 75.0), "[25.0, 50.0, 75.0]");
+  }
+
+  public void statsBinRunner(List<Double> splits) throws Exception {
+    statsBinRunner(splits, null);
+  }
+
+  public void statsBinRunner(List<Double> splits, String splitsName) throws Exception {
+    int bin = 0;
+    for(Double d : stats.getSortedValues()) {
+      StatisticsProvider provider = (StatisticsProvider)variables.get("stats");
+      if(bin < splits.size()) {
+        double percentileOfBin = provider.getPercentile(splits.get(bin));
+        if (d > percentileOfBin) {
```

Comment:
I see, so provider.getPercentile() actually takes a percentile value (as a number between 0 and 99.99) and returns a corresponding raw value?  That makes sense if so.  Please confirm.